unicodetools/src/main/java/org/unicode/xml/UcdXML.java

```diff
@@ -0,0 +1,825 @@
+                    ucdxmloutputtypes =
+                            new UCDXMLOUTPUTTYPE[] {
+                                UCDXMLOUTPUTTYPE.valueOf(
+                                        options[OUTPUT].value.toUpperCase(Locale.ROOT))
+                            };
+                } catch (Exception e) {
+                    throw new IllegalArgumentException(
+                            "Could not convert "
+                                    + options[OUTPUT].value
+                                    + " to one of [FLAT|GROUPED]");
+                }
+            }
+            if (options[OUTPUTFOLDER].doesOccur) {
+                try {
+                    destinationFolder =
+                            new File(
+                                    options[OUTPUTFOLDER].value
+                                            + getVersionString(ucdVersion, 3)
+                                            + "\\xmltest\\");
+                    if (!destinationFolder.exists()) {
+                        if (!destinationFolder.mkdir()) {
+                            throw new IOException();
+                        }
+                    }
+                } catch (Exception e) {
+                    throw new IllegalArgumentException(
+                            "Could not find or create " + options[OUTPUTFOLDER].value);
+                }
+            } else {
+                throw new IllegalArgumentException(
+                        "Missing command line option: --outputfolder (or -f)");
+            }
+
+        } catch (Exception e) {
+            System.err.println(e.getMessage());
+            System.exit(1);
+        }
+
+        if (ucdVersion != null && destinationFolder.exists()) {
+            for (UCDXMLOUTPUTRANGE ucdxmloutputrange : ucdxmloutputranges) {
+                for (UCDXMLOUTPUTTYPE ucdxmloutputtype : ucdxmloutputtypes) {
+                    System.out.println(
+                            "Building the "
+                                    + ucdxmloutputrange
+                                    + " "
+                                    + ucdxmloutputtype
+                                    + " UcdXML file for "
+                                    + ucdVersion);
+                    buildUcdXMLFile(
+                            ucdVersion, destinationFolder, ucdxmloutputrange, ucdxmloutputtype);
+                }
+            }
+            System.out.println("End");
+            System.exit(0);
+        } else {
+            System.err.println("Unexpected error when building UcdXML file.");
+            System.exit(1);
+        }
+    }
+
+    private static void buildUcdXMLFile(
+            VersionInfo ucdVersion,
+            File destinationFolder,
+            UCDXMLOUTPUTRANGE outputRange,
+            UCDXMLOUTPUTTYPE outputType)
+            throws IOException, TransformerConfigurationException, SAXException {
+        int lowCodepoint = 0x0;
+        int highCodepoint = 0x10FFFF;
```

Comment:
nit: "code point" is two words, so it would be nice to change "...odepoint" to "...odePoint" / in this whole new package

Reply:
Done